get_connection/get_connection.go

```diff
@@ -0,0 +1,57 @@
+package get_connection
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type Connection struct {
+	Id   string `json:"id"`
+	Name string `json:"name"`
+}
+
+func GetConnection(url, authToken, connectionName string) (*string, error) {
+
+	req, _ := http.NewRequest("GET", url, nil)
+	req.Header.Add("authorization", fmt.Sprintf("Bearer %s", authToken))
+	res, err := http.DefaultClient.Do(req)
+	if res.StatusCode != http.StatusOK {
+		err := fmt.Errorf("expected %d got %d: \n%s", http.StatusOK, res.StatusCode, res.Body)
+		return nil, err
+	}
+
+	defer res.Body.Close()
+
+	body, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	conns, err := handleData(body)
+	if err != nil {
+		return nil, err
+	}
+
+	return findConnectionByName(conns, connectionName), nil
+}
+
+func findConnectionByName(ConnectionSlice []Connection, ConnectionName string) (connectionId *string) {
+
+	for _, c := range ConnectionSlice {
+		if c.Name == ConnectionName {
+			connectionId = &c.Id
+			break
```

Comment:
if you return in a `for` you get the same result, e.g. `return &c.Id` without the need to `break`.